stock_picking_show_return/tests/test_stock_picking_show_return.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2014-2017 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests import common
+
+
+@common.at_install(False)
+@common.post_install(True)
+class TestStockPickingShowReturn(common.TransactionCase):
+    def setUp(self):
+        super(TestStockPickingShowReturn, self).setUp()
+        self.product = self.env['product.product'].create({
+            'name': 'Test product',
+        })
+        picking_type = self.env.ref('stock.picking_type_internal')
+        self.picking = self.env['stock.picking'].create({
+            'picking_type_id': picking_type.id,
+            'location_id': picking_type.default_location_src_id.id,
+            'location_dest_id': picking_type.default_location_dest_id.id,
+            'move_lines': [(0, 0, {
+                'name': self.product.name,
+                'product_id': self.product.id,
+                'product_uom': self.product.uom_id.id,
+                'product_uom_qty': 1,
+            })],
+        })
+
+    def test_returned_ids_field(self):
+        self.picking.action_assign()
+        self.picking.move_line_ids.qty_done = 1
+        self.picking.action_done()
+        wizard = self.env['stock.return.picking'].with_context(
+            active_ids=self.picking.ids, active_id=self.picking.id).create({})
+        wizard.create_returns()
+        self.assertTrue(self.picking.returned_ids)
```

Comment:
@bizzappdev IMO here use like `self.assertTrue(bool(self.picking.returned_ids))`, It's better for assert True

Reply:
[](https://docs.python.org/2/library/unittest.html#unittest.TestCase.debug)
it is checking that automatically by itself. no value added from changing code.

Reply:
@bizzappdev yes, but if you checked with `bool` (`assertTrue`) then the result must be bool, that why I asking to only change  like this `bool(self.picking.returned_ids)`

Reply:
https://docs.python.org/2/library/unittest.html#unittest.TestCase.debug

Method | Checks that | New in
assertTrue(x) | bool(x) is True

the parameter value can be anything. it will automatically convert it to bool after that only it will check.
check the official documentation for the same.

Reply:
yep, thus I'd compare `len()` or at least `.ids`

Reply:
due to converting to bool wouldn't change the issue behavior you're describing

Reply:
> yep, thus I'd compare `len()` or at least `.ids`

Yes, but better to `bool` :smile:

Reply:
> @bizzappdev @alexey-pelykh I think that an empty object will return True So that assertTrue seems wrong, it will always be True.
Naa empty will not return True 
as mentioned by @alexey-pelykh 
https://github.com/python/cpython/blob/44a79cc5b3d1fb0c03c99077aa26def85ec26c67/Lib/unittest/case.py#L729

it has if condition and if the object is empty it will go in the If condition

Reply:
https://github.com/odoo/odoo/blob/b234af04c1226bd5d5fb28fae7087b7f130e53d4/odoo/models.py#L4907

`if recordset:` is same as `if bool(recordset):` fyi

Reply:
No need for `bool`, `len` or whatever. ORM recordsets are already evaluated as falsy if no records, and trueish otherwise, so `self.assertTrue(self.picking.returned_ids)` is perfect as is.

Reply:
@pedrobaeza Thanks for clarifying my confusion!

Reply:
Thanks also to @alexey-pelykh that was saying the same with links to ORM code :wink: